# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/loader/ModuleLoaderBase.h
# Commit: 1b987903eab3
# Full Hash: 1b987903eab3fd39c65e6cfd87e3027eb10e616b
# Author: Yoshi Cheng-Hao Huang <allstars.chh@gmail.com>
# Date: 2025-09-24 18:03 +0000
# Regressor Bug: 1968890
# File Overlap Count: 5
# Overlapping Files: js/loader/ModuleLoadRequest.cpp, js/loader/ModuleLoaderBase.cpp, js/loader/ModuleLoadRequest.h, js/loader/ModuleLoaderBase.h, dom/script/ScriptLoader.cpp
# Description:
#   Bug 1968890 - Part 7: Add Cancel method. r=jonco
# ==============================================================================

diff -r 75ae330e1b90 -r 1b987903eab3 js/loader/ModuleLoaderBase.h
--- a/js/loader/ModuleLoaderBase.h	Wed Sep 24 18:03:30 2025 +0000
+++ b/js/loader/ModuleLoaderBase.h	Wed Sep 24 18:03:30 2025 +0000
@@ -574,6 +574,7 @@
 
   void OnFetchSucceeded(ModuleLoadRequest* aRequest);
   void OnFetchFailed(ModuleLoadRequest* aRequest);
+  void Cancel(ModuleLoadRequest* aRequest);
 
   // The slot stored in ImportMetaResolve function.
   enum class ImportMetaSlots : uint32_t { ModulePrivateSlot = 0, SlotCount };
